src/goto-harness/recursive_initialization.cpp

```diff
@@ -472,35 +472,54 @@ std::string recursive_initialization_configt::to_string() const
   return out.str();
 }
 
-irep_idt recursive_initializationt::get_fresh_global_name(
-  const std::string &symbol_name,
-  const exprt &initial_value) const
+static symbolt &get_fresh_global_symbol(
+  symbol_tablet &symbol_table,
+  const std::string &symbol_base_name,
+  typet symbol_type,
+  irep_idt mode)
 {
+  source_locationt source_location{};
+  source_location.set_file(CPROVER_PREFIX "harness.c");
```

Comment:
:pick: magic string "harness" should probably be turned into a macro or constexpr